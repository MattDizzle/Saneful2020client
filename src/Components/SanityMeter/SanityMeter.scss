.barContainer {
  width: 100%;
  background-color: #5F5E7D;
  margin: 2px 0;
  border: 3px solid darkgray;
  border-radius: 11px;
  border-style: ridge;
}

.barLabel {
  color: white;
  font-size: 0.55rem;
  margin-right: 5px;
}

.title-barContainer{
  display: flex;
}


@media only screen and (min-width: 520px) {
  .barLabel {
    font-size: 0.8rem;
  }

  .barContainer {
    width: 100%;
    background-color: #5F5E7D;
    margin: 5px 0;
    border: 4px solid darkgray;
    border-radius: 11px;
    border-style: ridge;
  }
}
