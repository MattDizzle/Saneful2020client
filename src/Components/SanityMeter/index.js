import React from "react";

import './SanityMeter.scss';

const SanityMeter = (props) => {
  const { currentSanity } = props;

  const fillerStyles = {
    height: '100%',
    borderRadius: '10px',
    transition: '0.25s',
    width: `${currentSanity}%`,
    backgroundColor: '#250cc7',
    textAlign: 'right'
  };

  return (
    //entire bar
    //completed percentage
    <div className="title-barContainer">
      <p>Sanity:</p>
      <div className="barContainer">
        <div style={fillerStyles}>
          <span className="barLabel">{`${currentSanity}`}</span>
        </div>
      </div>
    </div>
  );
};

export default SanityMeter;