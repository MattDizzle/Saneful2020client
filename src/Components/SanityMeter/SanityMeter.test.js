import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import SanityMeter from "./index";

describe(`SanityMeter Component`, () => {
  describe(`Smoke test`, () => {
    it(`Renders without crashing`, () => {
      const div = document.createElement("div");
      ReactDOM.render(
        <BrowserRouter>
          <SanityMeter />
        </BrowserRouter>,
        div
      );
      ReactDOM.unmountComponentAtNode(div);
    });
  });

  describe(`Snapshot test`, () => {
    it(`Renders the UI as expected`, () => {
      const tree = renderer
        .create(
          <BrowserRouter>
            <SanityMeter />
          </BrowserRouter>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
