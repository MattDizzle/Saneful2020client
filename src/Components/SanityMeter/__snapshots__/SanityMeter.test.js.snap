// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SanityMeter Component Snapshot test Renders the UI as expected 1`] = `
<div
  className="title-barContainer"
>
  <p>
    Sanity:
  </p>
  <div
    className="barContainer"
  >
    <div
      style={
        Object {
          "backgroundColor": "#250cc7",
          "borderRadius": "10px",
          "height": "100%",
          "textAlign": "right",
          "transition": "0.25s",
          "width": "undefined%",
        }
      }
    >
      <span
        className="barLabel"
      >
        undefined
      </span>
    </div>
  </div>
</div>
`;
